packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { DataObjectWithCounter, dataObjectWithCounterFactory } from "./dataObjectWithCounter";
+import { allFactories } from "./testDataObjects";
+
+const childKey = "childKey";
+export class DataObjectWithChildDataObject extends DataObjectWithCounter {
+    private unrefTimeStampMs?: number;
+    private oldChild?: DataObjectWithCounter;
+    private child?: DataObjectWithCounter;
```

Comment:
Consider making child a getter that always returns a value, i.e., asserts if child does not exist. This has couple of advantages:
1. You don't have to add validations whenever this is accessed.
2. You ensure that no code uses this before it exists. Currently, if this happens, the test will silently continue but fail at some point later making it harder to debug.

Reply:
Child could be referenced or unreferenced, when we stop the child, it's simpler to write `this.child?.stop();` The only time this.child needs to be referenced is when we are unreferencing and starting the child (starting the child is a different PR discussion)

Reply:
Are you suggesting we always reference a new child when we have unreferenced the current one? This would seem to complicate things.

Reply:
I was thinking a singular interesting GC event would just be to create and reference, reference previous, and unreference as it is simpler to verify just those singular actions rather than a series of these steps.

Reply:
That way a single client can be singularly focused on having an unreferenced dataObject and watch it get GCed

Reply:
My observation was that there will always be a child which is active. Hence the comment about making this a required property.

Reply:
Ok, let's leave that for a later PR, I'll need to think more deeply about how we want to guarantee that a child is always active. I'll add a comment for now